service/history/workflow/mutable_state_impl.go

```diff
@@ -424,6 +425,10 @@ func NewMutableStateFromDB(
 		}
 	}
 
+	// Proto serialization replaces empty maps with nils, ensure this map is never nil.
+	if mutableState.executionInfo.SubStateMachinesByType == nil {
+		mutableState.executionInfo.SubStateMachinesByType = make(map[int32]*persistencespb.StateMachineMap)
+	}
```

Comment:
This should be done by the persistence serializer. We have something similar in the ShardInfoFromBlob method.

Reply:
Oh thanks, I didn't know we had that.